frontend/src/features/admin-form/create/logic/components/LogicContent/EditCondition/EditConditionBlock.tsx

```diff
@@ -0,0 +1,353 @@
+      resetField(`${name}.value`, { defaultValue: '' })
+      resetField(`${name}.state`)
+    }
+  }, [hasFieldIdChanged, name, resetField])
+
+  useEffect(() => {
+    if (!currentSelectedField) {
+      resetField(`${name}.ifValueType`)
+      return
+    }
+    setValue(
+      `${name}.ifValueType`,
+      getIfLogicType({
+        fieldType: currentSelectedField.fieldType,
+        conditionState: conditionStateValue,
+      }),
+    )
+  }, [conditionStateValue, currentSelectedField, name, resetField, setValue])
+
+  const items = useMemo(() => {
+    if (!logicableFields) return []
+
+    // Get subset of logicableFields that have not already been set to show on
+    // other logic conditions.
+    let subsetLogicableFields = logicableFields
+    if (logicTypeValue === LogicType.ShowFields) {
+      const thenValuesSet = new Set(showValueWatch.value)
+      subsetLogicableFields = pickBy(
+        subsetLogicableFields,
+        (f) => !thenValuesSet.has(f._id),
+      )
+    }
+
+    return Object.entries(subsetLogicableFields).map(([key, value]) => ({
+      label: `${value.questionNumber}. ${value.title}`,
+      value: key,
+      icon: BASICFIELD_TO_DRAWER_META[value.fieldType].icon,
+    }))
+  }, [logicableFields, logicTypeValue, showValueWatch])
+
+  const conditionItems = useMemo(() => {
+    if (!currentSelectedField) return []
+    return (
+      LOGIC_MAP.get(currentSelectedField.fieldType)?.map((v) => ({
+        // Remove leading 'is' from the condition name for rendering.
+        label: v.replace(/^is\s/i, ''),
+        value: v,
+      })) ?? []
+    )
+  }, [currentSelectedField])
+
+  const conditionValueItems = useMemo(() => {
+    if (!ifFieldIdValue || !mapIdToField) return []
+    const mappedField = mapIdToField[ifFieldIdValue]
+    if (!mappedField) return []
+    switch (mappedField.fieldType) {
+      case BasicField.YesNo:
+        return ['Yes', 'No']
+      case BasicField.Radio:
+      case BasicField.Dropdown:
+        return mappedField.fieldOptions
+      case BasicField.Rating:
+        return range(1, mappedField.ratingOptions.steps + 1).map(String)
+      default:
+        return []
+    }
+  }, [ifFieldIdValue, mapIdToField])
+
+  const logicTypeWrapperWidth = useMemo(() => {
+    if (!currentSelectedField) return '9rem'
+    switch (currentSelectedField.fieldType) {
+      case BasicField.Dropdown:
+      case BasicField.Radio:
+      case BasicField.YesNo:
+        return '9rem'
+      default:
+        return '14rem'
+    }
+  }, [currentSelectedField])
+
+  const renderValueInputComponent = useCallback(
+    (field: ControllerRenderProps<EditLogicInputs, `${typeof name}.value`>) => {
+      const selectProps = {
+        isDisabled: !conditionStateValue || isLoading,
+        isSearchable: false,
+        placeholder: null,
+        isClearable: false,
+        items: conditionValueItems,
+      }
+      const { value, ...rest } = field
+
+      if (!ifValueTypeValue) {
+        return (
```

Comment:
out of curiosity, does TS complain if you use `case undefined` for this in the `switch` statement below?

Reply:
TIL you can use `undefined` as a case. It works fine.